### PR TITLE
[DC-915] Fixes to annotations and transform code

### PR DESCRIPTION
**Pull request by bentrahan**
TL;DR -- only look at changes to `util/transform_training.py`.

This PR covers two things, for which I am very very (very) sorry.  One is some hand-fixes to the annotated data, which had oddball syntax bugs in various places.  Probably due to some flavor of bug in the annotation tool.  You can safely ignore all of these.

The second and more meaningful fix is replacing the old regex-based format converter with a parser combinator so that we can correctly capture nested tags.  Previously we had to hand-fix annotation errors derived from incorrectly converting nested tags.  Since OpenNLP doesn't appear to support nested tags the combinator is set up to simply throw the inner tags on the floor and convert only the outer ones.  It also preserves whitespace to the extent possible within the bounds of my profound laziness.

The YEDDA format looks like this:

1. `[@Human-annotated entity#Tagtype*]`
2. `[$Suggested entity#Tagtype*]`

The OpenNLP format looks like this:

```
  <START:Tagtype> Any tagged text <END>
```

### CODE REVIEW

File: util/transform_training.py [hunk 1]
```diff
@@ -18,9 +61,6 @@
     required=True,
     help="output directory to which to write transformed files")
 
-YEDDA_RE = re.compile(r"""\[[@$]([^#]+)#([^*]+)\*\]""")
-
-
 def main(args: argparse.Namespace) -> None:
```
Comment (lijifly) [inline util/transform_training.py]: what's the use of None here?

**Comment by lijifly** [comment]
Just has a minor question. In general looks good to me.

File: util/transform_training.py (hunk 1 above)
Comment (bentrahan) [reply util/transform_training.py]: I believe it's a type annotation indicating the method always returns `None`.  This code was originally written by @genos who knows far more about Python type annotations than I do...

Comment (lijifly) [reply util/transform_training.py]: Got it. Thanks.

**Comment by genos** [comment]
I wholeheartedly endorse parser combinators! Type annotations are a Python 3 extravagance that aren’t used for anything, but external tools like mypy can typecheck them for you. Anyway +1e7
